cmd/oci-image-tool/validate.go

```diff
@@ -67,6 +68,11 @@ func newValidateCmd(stdout, stderr *log.Logger) *cobra.Command {
 		`A set of refs pointing to the manifests to be validated. Each reference must be present in the "refs" subdirectory of the image. Only applicable if type is image or imageLayout.`,
 	)
 
+	cmd.Flags().BoolVarP(
+		&v.verbose, "verbose", "v", false,
+		fmt.Sprintf(`Output verbosely, including all objects digest to be validated. If unset, oci-image-tool will only output a OK or failed result.`),
+	)
```

Comment:
On Thu, Sep 08, 2016 at 08:56:40PM -0700, Stephen Day wrote:
> > +	cmd.Flags().BoolVarP(
> > +		&v.verbose, "verbose", "v", false,
> > +		fmt.Sprintf(`Output verbosely, including all objects digest to be validated. If unset, oci-image-tool will only output a OK or failed result.`),
> > +	)
>
> @xiekeyang You don't unite these. Logging and output are different
> things. Typically, logging provides hints at how a program is doing,
> whereas the output is the primary purpose of the program.

I agree with this.

> Under a verbose flag, I would expect that to affect primary program
> output and not really have anything to do with logs, although this
> varies in practice.

This is surprising to me.  For example, ‘cp -v …’ prints hints about
what the program is doing.  Printing:

  ‘x’ -> ‘y’

is not the primary goal of the command (which is copying the files,
and not about writing anything to stdio).  Can you point to an example
where --verbose affects the primary program output?

In this case, the goal of the program is to validate a ref.  I expect
the primary purpose of the program is to output those PASS/FAIL test
results (e.g. in TAP [1], although we aren't using TAP here, more's
the pity ;).  A --verbose or --log-level option should have no effect
on that test output.  What it should do is adjust the log verbosity,
so we get information about the tests as they're being carried out.

This PR (as it stands with 4badd20d9) is sitting somewhere in the
middle.  If the digest being validated is a log output (which I think
it should be), it should be covered by --verbose or --log-level and
printed to stderr *when we start validating it* (4badd20d9 collects
digests and prints them after the test passes).  If the digest being
validated is part of the primary program output, we don't need
--verbose or --log-level, and should be printing it to stdout as a
sub-test of the ref validation.  In TAP, that would look like:

  TAP version 13
  ok 1 - manifest.v1+json sha256:abc…
  ok 2 - layer.tar+gzip sha256:def…
  ok 3 - ref v1 (sha256:abc…)
  ok 4 - manifest.v1+json sha256:123…
  ok 5 - layer.tar+gzip sha256:456…
  ok 6 - ref v2 (sha256:123…)
  1..6

[1]: https://testanything.org/